pkg/prometheus/instance/instance.go

```diff
@@ -614,3 +619,32 @@ func (vc *MetricValueCollector) GetValues(label string, labelValues ...string) (
 
 	return vals, nil
 }
+
+type runGroup struct {
+	cancel context.CancelFunc
+
+	g *run.Group
+}
+
+// newRunGroup wraps around run.Group but accepts a context. The run group will
+// be stopped if the context gets canceled.
+func newRunGroup(ctx context.Context) *runGroup {
```

Comment:
nit: `runGroupWithContext` or `runGroupWithCancel` might be better names that clash less with `run.Group`. wdyt?

Reply:
makes sense to me, changing it